### PR TITLE
Architecture chapters re-organization

### PR DESCRIPTION
**Pull request by mwasko**
Collect SOF Architecture chapters (both host Linux driver and firmware) in one Architecture tree.
The idea is to have architecture that describe current design and how things work in firmware and driver.
The developers guide should be limited to information how to implement or integrate new components, HAL, other.

### CODE REVIEW

File: architectures/firmware/index.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.. _architecture-firmware:
+
+Firmware Architecture
+################
```
Comment (greg-intel) [inline architectures/firmware/index.rst:4]: Was the title changed, but not the number of #?

File: architectures/host/linux_driver/architecture/sof_driver_arch.rst [hunk 2]
```diff
@@ -1,8 +1,18 @@
 .. _sof_driver_arch:
 
-SOF Driver Architecture
+SOF Linux Driver Architecture
 #######################
```
Comment (greg-intel) [inline architectures/host/linux_driver/architecture/sof_driver_arch.rst:4]: I believe...
```
SOF Linux Driver Architecture
#######################
```
Needs to be 
```
SOF Linux Driver Architecture
#############################
```
Per [RST syntax](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#headings)

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Approved on the basis that the minor comments above will be fixed.

**Comment by greg-intel** [comment]
I reran the checks, [build still failing](https://github.com/thesofproject/sof-docs/runs/4661741678?check_suite_focus=true) 
Is it possible something is checking for the files at previous locations?
```/home/runner/work/sof-docs/sof-docs/architectures/firmware/schedulers.rst:66: WARNING: PlantUML file "images/edf-scheduler-flow.pu" cannot be read: [Errno 2] No such file or directory: '/home/runner/work/sof-docs/sof-docs/architectures/firmware/images/edf-scheduler-flow.pu'```

**Comment by mwasko** [comment]
> I reran the checks, [build still failing](https://github.com/thesofproject/sof-docs/runs/4661741678?check_suite_focus=true) Is it possible something is checking for the files at previous locations? `/home/runner/work/sof-docs/sof-docs/architectures/firmware/schedulers.rst:66: WARNING: PlantUML file "images/edf-scheduler-flow.pu" cannot be read: [Errno 2] No such file or directory: '/home/runner/work/sof-docs/sof-docs/architectures/firmware/images/edf-scheduler-flow.pu'`

Fixed

File: developer_guides/add_new_arch.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
+.. _add_new_arch:
+
+Adding a new DSP architecture to SOF
+====================================
+
+This is not yet a guide for architecure porting, but in general are two ways to 
+add support for new DSP architectures to SOF.
```
Comment (anton-intel) [inline developer_guides/add_new_arch.rst:7]: This is not yet a guide for architecure porting, but in general, you can add support
for new DSP architectures to SOF in the following two ways:

File: developer_guides/add_new_arch.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+Adding a new architecture also usually means adding a new host platform too.
+
+Method 2 - Use existing RTOS
+----------------------------
+
+This method involves creating a HAL by wrapping the RTOS functions used by SOF
+as thinly as possible (i.e. to compile out). It also means removing unused code
+from the SOF build in order to use the RTOS version if desireable i.e.
+allocator, schedulers, messaging etc. The final stage is to link the SOF audio
+code to the RTOS.
```
Comment (anton-intel) [inline developer_guides/add_new_arch.rst:34]: This method involves creating a HAL by wrapping the RTOS functions used by SOF
as thinly as possible (that is to compile out). It also means removing unused code,
such as allocators, schedulers, and messaging, from the SOF build in order to use
the RTOS version, if desirable. The final stage is to link the SOF audio code to the RTOS.

File: architectures/firmware/overview.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
+.. _overview:
+
+Overview
+##########
+
+Currently SOF has support for the Cadence Xtensa DSP architecture in UP and SMP 
+modes in the upstream code base today.
```
Comment (anton-intel) [inline architectures/firmware/overview.rst:7]: Do we need to add "today" if we already say "currently"?

File: architectures/firmware/overview.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+The diagram below shows the high-level firmware architecture with the
+Baytrail platform integration as an example. The firmware is divided into four
```
Comment (anton-intel) [inline architectures/firmware/overview.rst:10]: Bay Trail

File: architectures/firmware/index.rst [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+   overview
+   mem-mgmt
+   pm-runtime/index
+   schedulers
+   drivers/index
+   components/index
+   pipelines/index
+   kd_integration/index
+
+Vendor Specific Architecture Information
```
Comment (anton-intel) [inline architectures/firmware/index.rst:18]: Vendor-Specific

File: architectures/firmware/index.rst [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+========================================
+
+Architecture details of any vendor specific code and flows. This is architecture
```
Comment (anton-intel) [inline architectures/firmware/index.rst:21]: vendor-specific

File: developer_guides/add_new_arch.rst [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+Adding a new DSP architecture to SOF
+====================================
+
+This is not yet a guide for architecure porting, but in general are two ways to 
+add support for new DSP architectures to SOF.
+
+#. Write a new Hardware Abstraction Layer (HAL) for your DSP.
+
+#. Use an existing RTOS that supports your DSP architecture as a HAL for SOF.
```
Comment (anton-intel) [inline developer_guides/add_new_arch.rst:11]: I'd use a bullet list here instead of a numbered list...

Comment (mwasko) [reply developer_guides/add_new_arch.rst:11]: @anton-intel the purpose of this commit is to re-organize place where content exist in chapters tree - not to refactor the content itself.

File: developer_guides/add_new_arch.rst (hunk 4 above)
Comment (mwasko) [reply developer_guides/add_new_arch.rst:33]: @anton-intel the purpose of this commit is to re-organize place where content exist in chapters tree - not to refactor the content itself.

**Review by greg-intel — APPROVED** [review]

**Pull request merged**